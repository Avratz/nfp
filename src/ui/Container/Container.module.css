.container {
  width: 100%;
}
.containerPadding {
  padding-left: 2rem;
  padding-right: 2rem;
}

@media screen and (min-width: 1200px) {
  .container {
    width: 1200px;
    margin-left: auto;
    margin-right: auto;
  }
}
