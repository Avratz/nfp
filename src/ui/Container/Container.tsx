import { cx } from 'class-variance-authority'
import { createElement } from 'react'

import styles from './Container.module.css'

type ContainerProps = {
  children: React.ReactNode
  as?: React.ElementType
  className?: string
  padding?: boolean
}

export function Container({
  children,
  className,
  padding = true,
  as = 'div',
}: ContainerProps) {
  return createElement(
    as,
    {
      className: cx(
        className,
        styles.container,
        padding ? styles.containerPadding : '',
      ),
    },
    children,
  )
}
