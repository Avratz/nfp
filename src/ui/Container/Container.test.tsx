import { render } from '@testing-library/react'
import { Container } from './Container'

describe('Container', () => {
  it('renders default', () => {
    const { container } = render(<Container>Container</Container>)
    expect(container).toMatchSnapshot()
  })

  it('renders as section with no padding', () => {
    const { container } = render(
      <Container as="section" padding={false}>
        section no padding
      </Container>,
    )
    expect(container).toMatchSnapshot()
  })
})
