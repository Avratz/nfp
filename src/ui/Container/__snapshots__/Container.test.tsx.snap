// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders as section with no padding 1`] = `
<div>
  <section
    class="container"
  >
    section no padding
  </section>
</div>
`;

exports[`Container renders default 1`] = `
<div>
  <div
    class="container containerPadding"
  >
    Container
  </div>
</div>
`;
